### PR TITLE
Remove spurious error queue entries on early data (3.1/3.0)

### PR DESCRIPTION
**Pull request by mattcaswell**
Early data decryption is expected to fail sometimes. If it does we should
not leave spurious error entries on the queue.

Fixes https://github.com/openssl/openssl/issues/20377

This is a backport of #20401 for 3.1/3.0.

### CODE REVIEW

**Comment by mattcaswell** [comment]
Ping - needs a second review

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**
